command_line/cosym.py

```diff
@@ -232,6 +234,19 @@ def _apply_reindexing_operators(self, reindexing_ops, subgroup=None):
                 )
             )
             refl["miller_index"] = cb_op.apply(refl["miller_index"])
+        # Allow for the case where some datasets are filtered out.
+        if len(reindexing_ops) < len(self._experiments):
+            to_delete = []
+            for i in range(len(self._experiments)):
+                if i not in unique_ids:
+                    to_delete.append(i)
+            to_delete.sort(reverse=True)
+            for idx in to_delete:
+                logger.info(
+                    f"Removing dataset {idx} as unable to determine reindexing operator"
+                )
+                del self._experiments[idx]
+                del self._reflections[idx]
```

Comment:
A slightly shorter, possibly cleaner, approach requiring only a single explicit loop:
```suggestion
        # Allow for the case where some datasets are filtered out.
        input_ids = set(range(len(self._experiments)))
        to_delete = sorted(input_ids - unique_ids)
        for idx in to_delete:
            logger.info(
                f"Removing dataset {idx} as unable to determine reindexing operator"
            )
            del self._experiments[idx]
            del self._reflections[idx]
```